File: datasets/IR-Plag/convertai-original-java-rest-typescript/case-07/plagiarized/L3/08/Level3.ts
import { Scanner } from 'readline-sync';

function diagonal(x: number[][]): number {
    let jumlah = 0;
    for (let i = 0; i < x.length; i++) {
        jumlah += x[i][i];
    }
    return jumlah;
}

function main(): void {
    const scan = new Scanner();
    const matriks: number[][] = new Array(4).fill(0).map(() => new Array(4).fill(0));

    console.log('Enter a 4 by 4 matrix row by row: ');
    for (let bar = 0; bar < 4; bar++) {
        for (let kol = 0; kol < 4; kol++) {
            matriks[bar][kol] = scan.nextDouble();
        }
    }
    console.log(`Sum of the elements in the major diagonal is ${diagonal(matriks)}`);
}

main();

File: datasets/IR-Plag/convertai-original-java-rest-kotlin/case-07/plagiarized/L3/08/Level3.kt
package IR

import java.util.Scanner

/**
 * @author FD8DBE9073AFCC0504CD6901E1091CAD
 */
object Level3 {
    @JvmStatic
    fun main(args: Array<String>) {
        val scan = Scanner(System.`in`)
        val matriks = Array(4) { DoubleArray(4) }

        print("Enter a 4 by 4 matrix row by row: ")
        for (bar in 0 until 4) {
            for (kol in 0 until 4) {
                matriks[bar][kol] = scan.nextDouble()
            }
        }
        println("Sum of the elements in the major diagonal is ${diagonal(matriks)}")
    }

    private fun diagonal(x: Array<DoubleArray>): Double {
        var jumlah = 0.0
        for (i in x.indices)
            jumlah += x[i][i]
        return jumlah
    }
}

